src/main/java/appeng/parts/reporting/AbstractPartReporting.java

```diff
@@ -172,16 +172,21 @@ public void writeToStream( final ByteBuf data ) throws IOException
 		}
 
 		data.writeByte( (byte) this.getClientFlags() );
+		data.writeFloat( this.opacity );
 	}
 
 	@Override
 	public boolean readFromStream( final ByteBuf data ) throws IOException
 	{
 		super.readFromStream( data );
 		final int oldFlags = this.getClientFlags();
+		final float oldOpacity = this.opacity;
+
 		this.clientFlags = data.readByte();
+		this.opacity = data.readFloat();
+
 		this.spin = (byte) ( this.getClientFlags() & 3 );
-		if( this.getClientFlags() == oldFlags )
+		if( this.getClientFlags() == oldFlags && this.opacity == oldOpacity )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Equality tests should not be made with floating point values. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1244)